extra_data/components.py

```diff
@@ -334,6 +344,20 @@ def get_array(self, key, pulses=np.s_[:], unstack_pulses=True):
           all pulses. Only used for per-train data.
         unstack_pulses: bool
           Whether to separate train and pulse dimensions.
+        fill_value: int or float, optional
+            Value to use for missing values. If None (default) the fill value
+            is 0 for integers and np.nan for floats.
+        join: str, optional
+            String indicating how to combine detector modules in the new array:
+            - “outer”: use the union of object indexes
+            - “inner”: use the intersection of object indexes
+            - “left”: use indexes from the first object with each dimension
+            - “right”: use indexes from the last object with each dimension
+            - “exact”: instead of aligning, raise ValueError when indexes to be
+                aligned are not equal
+            - “override”: if indexes are of same size, rewrite indexes to be
+                those of the first object with that dimension. Indexes for the
+                same dimension must have the same size in all objects.
```

Comment:
I don't think this description from Xarray is very helpful for someone using EXtra-data.

We can already control inner vs outer join by setting `min_modules` when creating the detector object - `min_modules=16` gives us the same effect as an inner join, `min_modules=1` (default) gives us an outer join. I'm not sure anyone would ever use the other options in this context, so maybe we don't need this parameter after all.